src/nodes/types.ts

```diff
@@ -6,9 +6,12 @@ import type { Host, Hostname, Port } from '../network/types';
  */
 type NodeGraphSpace = '0' | '1';
 
+type NodeAddressScope = 'local' | 'external';
+
 type NodeAddress = {
   host: Host | Hostname;
   port: Port;
+  scopes: Array<NodeAddressScope>;
 };
```

Comment:
Adding in the scopes field makes this a little clunky. But this seems fine for now. We can go over this again with the node graph expansion.